client/src/docs-ui/component-playground/component-playground.tsx

```diff
@@ -0,0 +1,21 @@
+import {Component, h, Prop} from "@stencil/core";
+import {AuthenticatorWithSlots} from "./custom-components/authenticator-with-slots";
+
+@Component({
+  tag: "docs-component-playground",
+})
+export class DocsComponentPlayground {
+  /** Name of component used in the playground */
+  @Prop() readonly componentName: string;
+
+  renderCustomComponent(componentName: string) {
+    switch (componentName) {
+      case "AuthenticatorWithSlots":
+        return AuthenticatorWithSlots();
+    }
+  }
+
+  render() {
+    return this.renderCustomComponent(this.componentName);
```

Comment:
Is it really worth extracting into another member?

Reply:
Yea, if we wanted to build an interactive playground around the custom component, we wouldn't have to extract a big switch statement.